frontend/ui/widget/container/framecontainer.lua

```diff
@@ -41,8 +41,17 @@ local FrameContainer = WidgetContainer:new{
     invert = false,
     allow_mirroring = true,
     _mirroredUI = BD.mirroredUILayout(),
+    focusable = false,
+    focus_border_size = Size.border.window * 2,
+    focus_border_color = Blitbuffer.COLOR_BLACK,
 }
 
+function FrameContainer:init()
+    WidgetContainer.init(self)
+    self.origin_bordersize = nil
+    self.origin_border_color = nil
+end
```

Comment:
(Didn't notice it earlier:)
It seems this :init() is not needed: you don't actually do anything in there.
These `self.origin_bordersize ` and `self.anything ` would be `nil` if you didn't have this :init() - just like most of the stuff on line 28 to 40 (which are there mostly for documentation purpose).
If these self.origin_bordersize are only internal, you don't even need to document them. Just assign to them when needed, `self.origin_border_color = self.bordersize`, no need for any previous declaration.

Reply:
Agreed.

Well, I was following best practices from Python and JavaScript experience which declaring internal variables explicitly in constructor. IDE may have better support(like code completion) by this pattern. Other developers can do a quick search to find out and learn well quickly such internal status lifecycle(initial value, how used, how changed).

Reply:
That's an internal variable? Please prefix it with _ then.

Reply:
And I'd say we tend to stick them in here, not in init unless there's something more complex involved.
 https://github.com/koreader/koreader/blob/eeb90c421e7bbc90cd7e38181a9e33841761b1e5/frontend/ui/widget/container/framecontainer.lua#L27-L47